### DISCUSSION TITLE
Load Amazon Sagemaker NTM model locally for inference

### DISCUSSION THREAD

**Discussion by ydennisy** [apache/incubator-mxnet#19258] [Q&A]
Hi!

I am very new to mxnet, using it for the first time  and having some issues loading an AWS Sagemaker trained model for local inference, I was earlier piggy backing off this [thread](https://github.com/apache/incubator-mxnet/discussions/16230
).

The main issue that I am having is that there is no error! My code runs fine, but the results are not as expected...

I am trying to load a model and predict on new data using the following code:

```python
sym, arg_params, aux_params = mx.model.load_checkpoint('model_algo-1', 0)
module_model = mx.mod.Module(symbol=sym, label_names=None, context=mx.cpu())

module_model.bind(
    for_training=False,
    data_shapes=[('data', (1, VOCAB_SIZE))]
)

# must set allow missing true here or receive an error for a missing n_epoch var
module_model.set_params(arg_params=arg_params, aux_params=aux_params, allow_missing=True)

module_model.predict(x) # where x is a numpy array of size (1, VOCAB_SIZE)
```

The code runs but the output is a single value rather than a distribution over topics as I would expect:
```
[11.060672]
<NDArray 1 @cpu(0)>
```

Would really appreciate some help/advice here!

**Comment by ydennisy** [comment]
**UPDATE:**

I have tried to load it using the Symbol API, but still no luck:
```python
import warnings
with warnings.catch_warnings():
    warnings.simplefilter('ignore')
    deserialized_net = gluon.nn.SymbolBlock.imports('model_algo-1-symbol.json', ['data'], 'model_algo-1-0000.params', ctx=mx.cpu())
```
Error:
```
AssertionError: Parameter 'n_epoch' is missing in file: model_algo-1-0000.params, which contains parameters: 'logsigma_bias', 'enc_0_bias', 'projection_bias', ..., 'enc_1_weight', 'enc_0_weight', 'mean_bias', 'logsigma_weight'. Please make sure source and target networks have the same prefix.
```

**Comment by lanking520** [comment]
You can try to visualize the graph and see how many inputs are there: 

https://mxnet.apache.org/versions/1.1.0/faq/visualize_graph.html

**Reply by ydennisy** [reply]
Thanks @lanking520 could you please elaborate a little - this take a network in the example - am I able to pass it one of my files?

Also I am able to manually view the files to look at the steps, sizes etc - but I am not sure what I am looking for.

**Comment by Zha0q1** [comment]
Hi Dennis, thanks for reaching out. I will look into this and get back to you soon

**Comment by Zha0q1** [comment]
Hi Dennis, would it be possible for you to provide the model files and input so that I can try to reproduce the issue? Also which doc did you follow?
